test/e2e/tests/settings/ipfs-ens-resolution.spec.js

```diff
@@ -1,190 +1,56 @@
-          id: (await req.body.getJson()).id,
-          result:
-            '0x0000000000000000000000000000000000000000000000000000000000000000',
-        },
-      };
-    });
-
-  // Content Hash Call
-  await mockServer
-    .forPost()
-    .withJsonBodyIncluding({
-      method: 'eth_call',
-      params: [
-        {
-          data: '0xbc1c58d1e650784434622eeb4ffbbb3220ebb371e26ad1a77f388680d42d8b1624baa6df',
-          to: ENS_PUBLIC_RESOLVER,
-        },
-      ],
-    })
-    .thenCallback(async (req) => {
-      return {
-        statusCode: 200,
-        json: {
-          jsonrpc: '2.0',
-          id: (await req.body.getJson()).id,
-          result:
-            '0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000',
-        },
-      };
-    });
-
-  // Token Balance Call
-  await mockServer
-    .forPost()
-    .withJsonBodyIncluding({
-      method: 'eth_call',
-      params: [
-        {
-          to: BALANCE_CHECKER,
-        },
-      ],
-    })
-    .thenCallback(async (req) => {
-      return {
-        statusCode: 200,
-        json: {
-          jsonrpc: '2.0',
-          id: (await req.body.getJson()).id,
-          result:
-            '0x0000000000000000000000000000000000000000000000000000000000000000',
-        },
-      };
-    });
-}
 
 describe('Settings', function () {
   const ENS_NAME = 'metamask.eth';
   const ENS_NAME_URL = `https://${ENS_NAME}/`;
   const ENS_DESTINATION_URL = `https://app.ens.domains/name/${ENS_NAME}`;
 
-  it('Redirects to ENS domains when user inputs ENS into address bar', async function () {
-    // Using proxy port that doesn't resolve so that the browser can error out properly
-    // on the ".eth" hostname.  The proxy does too much interference with 8000.
-
+  it(`Redirects to ENS domains when user inputs ENS into address bar`, async function () {
+    async function mockMetaMaskDotEth(mockServer) {
+      return await mockServer.forGet(ENS_NAME_URL).thenResetConnection();
+    }
+    async function mockEnsDotDomains(mockServer) {
+      return await mockServer
+        .forGet(ENS_DESTINATION_URL)
+        .thenReply(200, 'mocked ENS domain');
+    }
+    async function mockEns(mockServer) {
+      return [
+        await mockMetaMaskDotEth(mockServer),
+        await mockEnsDotDomains(mockServer),
+      ];
+    }
     await withFixtures(
       {
-        fixtures: new FixtureBuilder().withNetworkControllerOnMainnet().build(),
-        ganacheOptions: defaultGanacheOptions,
-        title: this.test.title,
-        testSpecificMock: mockInfura,
-        driverOptions: {
-          proxyPort: '8001',
-        },
+        title: this.test.fullTitle(),
+        testSpecificMock: mockEns,
       },
       async ({ driver }) => {
-        await driver.navigate();
+        // if the background/offscreen pages are ready metamask should be ready
+        // to handle ENS domain resolution
+        const page =
```

Comment:
Why do we need to navigate to one of these pages? Why can't we rely on the `controller-loaded` class being added to the ui?

Reply:
oh that's a great point! Taking a deeper look :eyeglasses:

Reply:
I see that the `controller-loaded` class is added in the UI after the `start` function is called. Unfortunately, this would get run before initialization has completed. That function triggers the start of UI initialization, but doesn't block on it finishing.

If it did wait for initialization to complete, that might work. UI initialization does depend upon background initialization already. Or it's supposed to at least, there's an event that we wait for that indicates when the background is ready.

Reply:
thank you for this clarifying comment, Mark! The `controller-load` class is now added after the initialization is complete. This fixed the issue and we can now rely on the UI class in the same way as we could rely on background or offscreen, for knowing the controllers are ready.

Thank you Dan for your initial comment! This has a great impact, since now all the tests will benefit from the correct state, where we look for the controllers-ready class in the UI in the `navigate` function